sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/EncryptColumnsTransformer.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.action.generic.transformer
+
+import com.typesafe.config.Config
+import io.smartdatalake.config.SdlConfigObject.{ActionId, DataObjectId}
+import io.smartdatalake.config.{FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.secrets.SecretsUtil
+import io.smartdatalake.workflow.ActionPipelineContext
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.expressions.UserDefinedFunction
+
+import java.security.SecureRandom
+import java.util.Base64
+import javax.crypto.spec.{GCMParameterSpec, SecretKeySpec}
+import javax.crypto.{Cipher, SecretKey}
+import org.apache.spark.sql.functions.{col, udf}
+
+trait EncryptDecrypt {
+  def key: Array[Byte] //= "test234".getBytes
+  val cryptUDF: UserDefinedFunction = udf(encrypt _)
+  private val ALGORITHM_STRING: String = "AES/GCM/PKCS5Padding"
+  private val IV_SIZE = 128
+  private val TAG_BIT_LENGTH = 128
+
+  private val secureRandom = new SecureRandom()
+
+
+  def process(df: DataFrame, encryptColumns: Seq[String]): DataFrame = {
+    var dfEnc = df
+    for (colName <- encryptColumns) {
+      dfEnc = dfEnc.withColumn(colName, cryptUDF(col(colName)))
+    }
+    dfEnc
+  }
```

Comment:
``
```suggestion
  def process(df: DataFrame, encryptColumns: Seq[String]): DataFrame = {
   
   encryptColumns.foldLeft(df){
   case (dfTemp, colName) => dfTemp.withColumn(colName, cryptUDF(col(colName)))
   }
  }
```
Try this! FoldLeft to the rescue. You can replace every loop with foldLefts. Complex and hierarchical loops get easier to read and you don t need evil variables